internal/pkg/service/common/servicectx/servicectx.go

```diff
@@ -79,8 +85,22 @@ func New(ctx context.Context, cancel context.CancelFunc, logger log.Logger, opts
 		wg:       &sync.WaitGroup{},
 		errCh:    errCh,
 		uniqueID: c.uniqueID,
+		lock:     &sync.Mutex{},
 	}
 
+	// Register onShutdown operation
+	proc.Add(func(ctx context.Context, errCh chan<- error) {
+		<-ctx.Done()
+		proc.lock.Lock()
+		proc.terminating = true
+		proc.lock.Unlock()
+
+		// Iterate callbacks in reverse order, LIFO
+		for i := len(proc.onShutdown) - 1; i >= 0; i-- {
+			proc.onShutdown[i]()
+		}
+	})
+
```

Comment:
Narazil som na to, ze `OnShutdown` callbacky potrebujem spustat sekvencne, v LastInFirstOut poradi.

Priklad:
- vytvorim etcd spojenie
- vytvorim etcd session pre rozdelenie prace medzi workers

Zatvarat to musim v opacnom poradi (LIFO):
- najprv sa ukonci session
- potom sa ukonci cele spojenie do etcd